include/types.h

```diff
@@ -988,6 +989,21 @@ typedef struct link_speed
 
 } link_speed_t;
 
+// handling gzip files
+
+typedef struct hc_fp
+{
+  union
+  {
+    FILE *fp;
+    gzFile gfp;
+  } f;
+
+  short is_gzip;
```

Comment:
Please change to bool

Reply:
If I change it to bool I can't check for "fp->is_gzip == -1". Please double check the hc_fopen function to understand what I mean.

Reply:
I was reviewing hc_fopen() but I do not see where you need to set -1. There's no mentioning of -1 in the function body at all.

Reply:
I remove it with the last patch, now it's bool and all is_gzip are true/false only